sklearn/utils/_plotting.py

```diff
@@ -63,6 +75,201 @@ def _validate_from_predictions_params(
+        **kwargs,
+    ):
+        """Get validated line kwargs for each curve.
+
+        Parameters
+        ----------
+        n_curves : int
+            Number of curves.
+
+        names : list[str]
+            Names of each curve.
+
+        summary_values : list[float]
+            List of summary values for each curve (e.g., ROC AUC, average precision).
+
+        summary_value_name : str
+            Name of the summary value provided in `summary_values`.
+
+        fold_line_kwargs : dict or list of dict
+            Dictionary with keywords passed to the matplotlib's `plot` function
+            to draw the individual ROC curves. If a list is provided, the
+            parameters are applied to the ROC curves sequentially. If a single
+            dictionary is provided, the same parameters are applied to all ROC
+            curves. Ignored for single curve plots - pass as `**kwargs` for
+            single curve plots.
+
+        default_line_kwargs : dict, default=None
+            Default line kwargs to be used in all curves, unless overridden by
+            `fold_line_kwargs`.
+
+        **kwargs : dict
+            For a single curve plots only, keyword arguments to be passed to
+            matplotlib's `plot`. Ignored for multi-curve plots - use `fold_line_kwargs`
+            for multi-curve plots.
+        """
+        # Ensure parameters are of the correct length
+        names_ = [None] * n_curves if names is None else names
+        summary_values_ = (
+            [None] * n_curves if summary_values is None else summary_values
+        )
+        # `fold_line_kwargs` ignored for single curve plots
+        # `kwargs` ignored for multi-curve plots
+        if n_curves == 1:
+            fold_line_kwargs = [kwargs]
+        else:
+            fold_line_kwargs = cls._validate_line_kwargs(n_curves, fold_line_kwargs)
+
+        if default_line_kwargs is None:
+            default_line_kwargs = {}
+        line_kwargs = []
+        for fold_idx, (curve_summary_value, curve_name) in enumerate(
+            zip(summary_values_, names_)
+        ):
+            if curve_summary_value is not None and curve_name is not None:
+                default_line_kwargs["label"] = (
+                    f"{curve_name} ({summary_value_name} = {curve_summary_value:0.2f})"
+                )
+            elif curve_summary_value is not None:
+                default_line_kwargs["label"] = (
+                    f"{summary_value_name} = {curve_summary_value:0.2f}"
+                )
+            elif curve_name is not None:
+                default_line_kwargs["label"] = curve_name
+
+            line_kwargs.append(
+                _validate_style_kwargs(default_line_kwargs, fold_line_kwargs[fold_idx])
+            )
+        return line_kwargs
+
+    # TODO: if useful for non binary displays (e.g.,`LearningCurveDisplay`,
+    # `ValidationCurveDisplay`) amend to function
+    @classmethod
+    def _validate_line_kwargs(
```

Comment:
I am wondering if it might be nice to test these class methods in isolation, to avoid repeating the same/similar tests in each child class - but may be it is better to test it in the child classes to make sure it works all together...?

cc @glemaitre

Reply:
We could test them. I think they should not be a class method but rather a static method because I don't see any use of cls